test/functional/p2p_v2.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test BIP324 v2 transport
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class P2PV2Test(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 5
+        self.extra_args = [["-v2transport=1"], ["-v2transport=1"], ["-v2transport=0"], ["-v2transport=0"], ["-v2transport=0"]]
+
+    def run_test(self):
+        sending_handshake = "start sending v2 handshake to peer"
+        downgrading_to_v1 = "retrying with v1 transport protocol for peer"
+        self.disconnect_nodes(0, 1)
+        self.disconnect_nodes(1, 2)
+        self.disconnect_nodes(2, 3)
+        self.disconnect_nodes(3, 4)
+
+        # V2 nodes can sync with V2 nodes
+        assert_equal(self.nodes[0].getblockcount(), 0)
+        assert_equal(self.nodes[1].getblockcount(), 0)
+        self.nodes[0].generatetoaddress(5, "bcrt1q0yq2azut8gn2xu3y2g0xucf8pny6w8uxmyf220", invalid_call=False)
+        assert_equal(self.nodes[0].getblockcount(), 5)
+        assert_equal(self.nodes[1].getblockcount(), 0)
+        with self.nodes[0].assert_debug_log(expected_msgs=[sending_handshake],
+                                            unexpected_msgs=[downgrading_to_v1]):
+            self.connect_nodes(0, 1, True)
+        # sync_all() verifies that the block tips match
+        self.sync_all(self.nodes[0:2])
```

Comment:
in commit be8a5dcddd36be8a5f137baf3ebd32d590a0747d: Somewhere around here it might make sense to verify that v2 is indeed used by checking the new `getpeerinfo` fields ("transport_protocol_type" and "session_id") for nodes 0 and 1.